### PR TITLE
Add 'Back to top' links to main and about pages

### PR DESCRIPTION
**Pull request by MikeCoats**
This PR adds a 'back to top' link to the main and about pages. It will appear once you've scrolled down by half a page and will stay overlayed on the screen until you hit the bottom of the page, where it then nestles between the content and the footer. It appears to work well at mobile resolutions and aspect ratios too.

# Screenshots

## No link at top of page

image

## Overlayed link when scrolled down page

image

## Inline link when scrolled down page
image

## No link at top of page on mobile

image

## Overlayed link when scrolled down page on mobile

image

## Inline link when scrolled down page on mobile

image

Fixes PublicHealthEngland/coronavirus-dashboard#97

### CODE REVIEW

File: src/components/BackToTop/BackToTop.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import useBackToTopOverlayVisible from 'hooks/useBackToTopOverlayVisible';
+
+const BackToTopLink: ComponentType<Props> = () => {
+  return (
+    <div class="govuk-width-container">
+      <a class="govuk-link govuk-link--no-visited-state" href="#top">
+        <svg role="presentation" focusable="false" class="back-to-top" xmlns="http://www.w3.org/2000/svg" width="13" height="17" viewBox="0 0 13 17">
+          <path fill="currentColor" d="M6.5 0L0 6.5 1.4 8l4-4v12.7h2V4l4.3 4L13 6.4z"></path>
+        </svg>Back to top
```
Comment (xenatisch) [inline src/components/BackToTop/BackToTop.js:17]: Nice touch!

File: src/App.js [hunk 2]
```diff
@@ -51,6 +52,10 @@ const App = () => {
                 homepageUrlHref="https://gov.uk"
             />
             <Navigation/>
+            <Switch>
+                <Route path="/about" exact render={()=>(<BackToTop mode="overlay"/>)} />
+                <Route path="/" exact render={()=>(<BackToTop mode="overlay"/>)} />
+            </Switch>
```
Comment (xenatisch) [inline src/App.js:58]: @MikeCoats Could you add a brief comment on this so devs won't get confused in the future?

File: src/components/BackToTop/BackToTop.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      </a>
+    </div>
+  );
+};
+
+const BackToTop: ComponentType<Props> = ({ mode }: Props) => {
```
Comment (xenatisch) [inline src/components/BackToTop/BackToTop.js:23]: @MikeCoats Maybe a brief doc for this function as well. Re using it as an overlay or otherwise.

**Comment by MikeCoats** [comment]
Yeah, no worries. I’ll fire in some more details when I’m back at my computer.

**Comment by xenatisch** [comment]
How are we doing with the docs / comments @MikeCoats? 

I'm hoping to include this PR in the next release tomorrow.

**Comment by MikeCoats** [comment]
I’ve done about half of the changes and I’ll finish the rest and push them tonight if that’s alright?

**Comment by MikeCoats** [comment]
@xenatisch That's the comments added. If they need a bit more clarification, let me know - I'll be WFH from about 8:15-8:30 tomorrow, and I'll update them as quick as I can.

**Comment by xenatisch** [comment]
@MikeCoats It's all good. Thank you! :)

**Comment by xenatisch** [comment]
[Add a "back to top" link](https://trello.com/c/UgTu9Dh7/253-add-a-back-to-top-link)

**Pull request merged**